File: __aacontest/1.py
class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        for i in range(len(s) - 1):
            substring = s[i:i+2]
            if substring in s[::-1]:
                return True
        return False

# Example usage:
solution = Solution()
print(solution.isSubstringPresent("leetcode"))  # Output: True
print(solution.isSubstringPresent("abcba"))     # Output: True
print(solution.isSubstringPresent("abcd"))      # Output: False


class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        return any(s[i:i+2][::-1] in s for i in range(len(s)-1))

File: __aacontest/1.rb
# @param {String} s
# @return {Boolean}
#s = "leetcode"
s = "abcba"
def is_substring_present(s)
  (s.chars.length-1).times do |i|
    return true if s[i..i+1] == s[i..i+1].reverse
  end
end
p is_substring_present(s)